examples/retry_command.ts

```diff
@@ -0,0 +1,34 @@
+import { Command, CommandStep, Pipeline } from '../src';
+import { YamlSerializer } from '../src/serializers/yaml';
+
+class RetryCommand extends Command {
+    constructor(public retries: number, public c: Command) {
+        super(c.toString());
+        this.retries = retries;
+    }
+
+    protected serialize(): string {
+        return new Array(this.retries + 1).fill(this.c).join(' || ');
+    }
+
+    public toString(): string {
+        return this.serialize();
```

Comment:
I think you want a human readable version here, e.g.:

```suggestion
        return `${this command} [retry = ${this.retries}]`
```